### PR TITLE
Add option to copy reader studies

### LINKED ISSUE #1245: Button to copy a reader study
Make a button to make a new reader study copying the values (but creating new entries) for:

- Reader study
  - User provided title
  - description, workstation, workstation config, logo, shuffle hanging list, help text
  - Images? hanging list?
- Questions
  - question text, help text, answer type, image port, required, direction, scoring function, order
  - Categorical option
    - title, default

### PR DESCRIPTION
**Pull request by MikeOverkamp-diag**
Closes #1245

### CODE REVIEW

File: app/grandchallenge/reader_studies/forms.py [hunk 1]
```diff
@@ -142,6 +143,20 @@ class Meta(ReaderStudyCreateForm.Meta):
         }
 
 
+class ReaderStudyCopyForm(Form):
+    copy_images = BooleanField(required=False)
+    copy_hanging_list = BooleanField(required=False)
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/forms.py:148]: We should do sanity checks - it makes no sense to copy the hanging list or case text without the images, etc.

Comment (jmsmkn) [reply app/grandchallenge/reader_studies/forms.py:148]: Or does it? Unsure.

File: app/grandchallenge/reader_studies/views.py [hunk 2]
```diff
@@ -405,6 +407,68 @@ def get_success_url(self):
...
+    reader_study = None
+
+    def get_permission_object(self):
+        return get_object_or_404(ReaderStudy, slug=self.kwargs["slug"])
+
+    def form_valid(self, form):  # noqa: C901
+        reader_study = self.get_permission_object()
+
+        rs = ReaderStudy.objects.create(
+            title=reader_study.title,
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/views.py:427]: Is there a better way to maintain the list of attributes that are copyable?

File: app/grandchallenge/reader_studies/forms.py [hunk 3]
```diff
@@ -142,6 +143,20 @@ class Meta(ReaderStudyCreateForm.Meta):
         }
 
 
+class ReaderStudyCopyForm(Form):
+    copy_images = BooleanField(required=False)
```
[the remark this answers is not in the document]
Comment (MikeOverkamp-diag) [reply app/grandchallenge/reader_studies/forms.py:147]: Currently false. Do you think true would be better?

File: app/grandchallenge/reader_studies/forms.py (hunk 1 above)
Comment (MikeOverkamp-diag) [reply app/grandchallenge/reader_studies/forms.py:148]: I'd say it makes little sense, a check would be good I think.

File: app/grandchallenge/reader_studies/forms.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmsmkn) [reply app/grandchallenge/reader_studies/forms.py:147]: I think the default should be `true` in that case.

File: app/grandchallenge/reader_studies/templates/reader_studies/readerstudy_copy.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+{% block breadcrumbs %}
+    <ol class="breadcrumb">
+        <li class="breadcrumb-item"><a href="{% url 'home' %}"><i
+                class="fas fa-home"></i></a></li>
+        <li class="breadcrumb-item"><a href="{% url 'reader-studies:list' %}">Reader
+            Studies</a></li>
+        {% if object %}
+            <li class="breadcrumb-item"><a
+                    href="{{ object.get_absolute_url }}">{{ object }}</a></li>
+        {% endif %}
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/templates/reader_studies/readerstudy_copy.html:15]: The object is missing, would be good to see which reader study you're copying from

File: app/grandchallenge/reader_studies/forms.py [hunk 5]
```diff
@@ -142,6 +143,30 @@ class Meta(ReaderStudyCreateForm.Meta):
         }
 
 
+class ReaderStudyCopyForm(Form):
+    copy_images = BooleanField(required=False, initial=True)
+    copy_hanging_list = BooleanField(required=False, initial=True)
+    copy_case_text = BooleanField(required=False, initial=True)
+    copy_questions = BooleanField(required=False, initial=True)
+    copy_readers = BooleanField(required=False, initial=True)
+    copy_editors = BooleanField(required=False, initial=True)
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/forms.py:152]: Can we set the title of the new reader study here too as this sets the slug, which is not editable by the user.

**Pull request merged**